src/Storages/MergeTree/BackgroundJobsExecutor.cpp

```diff
@@ -180,10 +180,16 @@ void IBackgroundJobExecutor::triggerTask()
 }
 
 void IBackgroundJobExecutor::backgroundTaskFunction()
+try
 {
     if (!scheduleJob())
         scheduleTask(/* with_backoff = */ true);
 }
+catch (...)
+{
+    tryLogCurrentException(__PRETTY_FUNCTION__);
```

Comment:
Could you please add comments? I did not get this change...

Reply:
Done, see also #27428

Reply:
> I still have difficulties understanding this fragment of code...

@alexey-milovidov You are talking about this try/catch, right?

This function is executed from the thread pool, and when job in the threadpool throws during execution this thread will be terminated, I can provide a stacktrace if this will help

Reply:
And why we do `scheduleTask` again?

Also worth adding comments around `scheduleJob` and `scheduleTask`, it is extremely confusing.

Reply:
>And why we do scheduleTask again?

To retry on exception, since it may be some temporary exception.

> Also worth adding comments around scheduleJob and scheduleTask, it is extremely confusing.

There are already comments in header, not enough?

Reply:
Ok. We need to reschedule this task to continue.